### PR TITLE
Show limitations of Build Matrix up front

### PR DESCRIPTION
**Pull request by plaindocs**

### CODE REVIEW

File: pages/pipelines/build_matrix.md.erb [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # Build Matrix
 
-Build Matrices lets you simplify complex build configurations by expanding a step template and array of matrix elements into multiple jobs.
+Build Matrices help you simplify complex build configurations by expanding a step template and array of matrix elements into multiple jobs.
```
Comment (ddbeck) [inline pages/pipelines/build_matrix.md.erb:3]: I know this was here already, but for consistency with the rest of the page:

```suggestion
Build matrices help you simplify complex build configurations by expanding a step template and array of matrix elements into multiple jobs.
```

File: pages/pipelines/build_matrix.md.erb [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+
+The following [command step](/docs/pipelines/command-step) attributes can contain matrix values for interpolation:
+
+* [environment variables](/docs/pipelines/environment-variables)
+* [labels](/docs/pipelines/command-step#label)
+* [commands](/docs/pipelines/command-step#command-step-attributes)
+
+Interpolation of matrix values inside other attributes, including step keys and plugin attributes is not supported.
```
Comment (ddbeck) [inline pages/pipelines/build_matrix.md.erb:11]: Feels like the most important thing here (the negation) is at the very, very end. Maybe flip it, like this?

```suggestion
You can't use matrix values in other attributes, including step keys and plugin attributes.
```

File: pages/pipelines/build_matrix.md.erb [hunk 3]
```diff
@@ -17,7 +25,7 @@ steps:
 ```
 {: codeblock-file="pipeline.yml"}
 
-You can use a Build Matrix to expand a single step into three steps by interpolating the matrix values into [command step](/docs/pipelines/command-step) attributes such as [environment variables](/docs/pipelines/environment-variables), labels and commands. Interpolation of matrix values inside plugin attributes is not supported.
+Use a Build Matrix to expand a single step template into three steps by interpolating the matrix values into the following build configuration:
```
Comment (ddbeck) [inline pages/pipelines/build_matrix.md.erb:28]: 'nother consistency change

```suggestion
Use a build matrix to expand a single step template into three steps by interpolating the matrix values into the following build configuration:
```

**Review by ddbeck — APPROVED** [review]
Looks good, Sam. I picked a few optional nits for you, though.

File: pages/pipelines/build_matrix.md.erb (hunk 2 above)
Comment (plaindocs) [reply pages/pipelines/build_matrix.md.erb:11]: LOL, can't believe I wrote that.

**Pull request merged**